oauth2_provider/oauth2_validators.py

```diff
@@ -240,13 +240,25 @@ def validate_client_id(self, client_id, request, *args, **kwargs):
     def get_default_redirect_uri(self, client_id, request, *args, **kwargs):
         return request.client.default_redirect_uri
 
-    def _get_token_from_authentication_server(self, token, introspection_url, introspection_token):
-        bearer = "Bearer {}".format(introspection_token)
+    def _get_token_from_authentication_server(self,
+                                              token,
+                                              introspection_url,
+                                              introspection_token,
+                                              introspection_credentials):
+        # some rfc7662 implementations use a Bearer token while others use Basic Auth
+        headers = None
+        if introspection_token:
+            headers = {"Authorization": "Bearer {}".format(introspection_token)}
+        elif introspection_credentials and len(introspection_credentials) == 2:
```

Comment:
What's with the `len(introspection_credentials) == 2`?

Reply:
introspection_credentials is expected to be a tuple or list of (client_id, client_secret). Just making sure the user provided only 2 items. I guess I should do a type check too since they could provide a string of length 2.... Or expect a dict keyed by `client_id` and `client_secret`.  What do you think?

Reply:
Not sure it's worth checking the format validity. It'll break if users pass the wrong format and that's Python for you. In 3.6 we'll add typings.